frontend/controller/actions/identity-kv.js

```diff
@@ -115,6 +115,30 @@ export default (sbp('sbp/selectors/register', {
       }
     })
   },
+  'gi.actions/identity/kv/markAsUnread': ({ contractID, messageHash, createdHeight, unreadMessages }: {
+    contractID: string,
+    messageHash: string,
+    createdHeight: number,
+    unreadMessages: Array<{ messageHash: string, createdHeight: number }>
+  }) => {
+    return sbp('okTurtles.eventQueue/queueEvent', KV_QUEUE, async () => {
+      const getUpdatedUnreadMessages = async () => {
+        const currentData = await sbp('gi.actions/identity/kv/fetchChatRoomUnreadMessages')
+
+        return {
+          ...currentData,
+          [contractID]: {
+            readUntil: { messageHash, createdHeight }, unreadMessages
+          }
+        }
+      }
+
+      const data = await getUpdatedUnreadMessages()
+      if (data) {
```

Comment:
The way the other unread actions are written check for `if (data)` because `getUpdatedUnreadMessages` could actually return `null`, but the way this version of it is written, it will never return `null`. So either this `if` should be removed, or `null` should be returned in some special case of `getUpdatedUnreadMessages`.  I have not looked closely enough to decide which should be the case (so please investigate), just noticing this inconsistency.

Reply:
@taoeffect Right. In this case, we can add a logic to check if the existing message hash in `readUntil` property is the same as the one requested (eg. User trying to mark the same message as unread multiple times.)